### DISCUSSION TITLE
Enabling fzf to paste the absolute path of selected files into the command line.

### DISCUSSION THREAD

**Discussion by yrammos** [junegunn/fzf#3307] [Q&A] [answered]
I like to maintain a command-line history with absolute file and directory paths (i.e.. `realpath`s), not relative paths, so that any commands that I retrieve from history can be executed from anywhere, in principle.

I'm interested in hearing how others would go about this. I'd rather not override the fzf widgets code, unless entirely necessary. Any hidden options or less invasive workarounds, perhaps?

Many thanks.

**Comment by Ioulan** [comment, marked as answer]
You can set it on your environment variables,

1. First solution:
If you're using [fd](https://github.com/sharkdp/fd), there's this `--absolute-path` flag e.g:
```sh
if command -v fd >/dev/null 2>&1; then
    export FZF_DEFAULT_COMMAND="fd --absolute-path --color=always --hidden --exclude .git"
else
    export FZF_DEFAULT_COMMAND="find ~+ -type f \! \( -path '*/.git' -prune \) -printf '%P\n'"
fi
```

2. Another solution (since on first one, the display results are very long, because of the absolute path), 
Now on the `Ctrl-t` widget pressing enter would append the $PWD/{selected line buffer}
```sh
export FZF_CTRL_T_OPTS="--bind 'enter:execute(echo $PWD/{$BUFFER})+abort'"
```

**Reply by yrammos** [reply]
Perfect — thank you!
